### PR TITLE
Storing information about when a user read messages from a specific c…

### PR DESCRIPTION
**Pull request by marcinJer**

### CODE REVIEW

File: src/main/java/com/example/chat/groupchatbackend/ReadStatus.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import javax.persistence.Id;
+import java.time.LocalDateTime;
+
+@Entity
+public class ReadStatus {
+
+    @Id
+    @GeneratedValue
+    private int id;
+    private int conversationId;
```
Comment (michal-marciniec) [inline src/main/java/com/example/chat/groupchatbackend/ReadStatus.java]: this won't generate foreign key between Conversation and ReadStatus. try something like here: https://stackoverflow.com/a/6311954

File: src/main/java/com/example/chat/groupchatbackend/ReadStatus.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    private int userId;
```
Comment (michal-marciniec) [inline src/main/java/com/example/chat/groupchatbackend/ReadStatus.java]: same for User

File: src/main/java/com/example/chat/groupchatbackend/controllers/MessagesController.java [hunk 3]
```diff
@@ -24,19 +25,32 @@
...
+            updateReadStatus(conversation, LocalDateTime.now());
             return getMessagesByDate(conversation, LocalDateTime.MIN);
         } else {
             LocalDateTime date = Instant.ofEpochMilli(timestamp).atZone(ZoneId.systemDefault()).toLocalDateTime();
+            updateReadStatus(conversation, date);
             return getMessagesByDate(conversation, date);
         }
     }
 
+    private void updateReadStatus(Conversation conversation, LocalDateTime date) {
```
Comment (michal-marciniec) [inline src/main/java/com/example/chat/groupchatbackend/controllers/MessagesController.java:49]: no more comments to the code, one thing to the funtionality - please test that when there is already a ReadStatus row for this Conversation and User , we don't want to add a new one, but update dateTime of the existing one

Comment (michal-marciniec) [reply src/main/java/com/example/chat/groupchatbackend/controllers/MessagesController.java:49]: actually we could do this automatically if we removed "id" field for ReadStatus and make `conversationId + userID` combination as our primary key

**Review by michal-marciniec — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/chat/groupchatbackend/controllers/MessagesController.java (hunk 3 above)
Comment (marcinJer) [reply src/main/java/com/example/chat/groupchatbackend/controllers/MessagesController.java:49]: It's not possible :smile:

**Review by michal-marciniec — APPROVED** [review]

**Review by alekott — APPROVED** [review]

**Comment by michal-marciniec** [comment]
@marcinJer ping - blocking tasks have been merged, so you can update this branch with fresh master and retest it

**Comment by michal-marciniec** [comment]
will be merged in #27

**Pull request closed without merging**